### ISSUE TITLE
Move HashContainerUtils.java from "templates" to "java" [HPPC-90]

### ISSUE THREAD

**Issue by dweiss** [carrotsearch/hppc#133]
Move HashContainerUtils.java from "templates" to "java".  More specifically:

    git mv hppc-core/src/main/templates/com/carrotsearch/hppc/HashContainerUtils.java hppc-core/src/main/java/com/carrotsearch/hppc/HashContainerUtils.java

Short and simple reason for this change:  The file doesn't contain anything for the template processor, so it doesn't belong in templates directory.

Long and complicated reason for this change:  I'm working on a not-yet-released Ant script to build hppc, and due to a bug in the script it didn't send the template source files to javac, only the template output files.  I discovered that, surprisingly, hppc compiles perfectly okay with only the template output files, if you just move HashContainerUtils as suggested!

I'll post a link to the commit on github in a moment.  (I want to submit this first so I can get an issue number, then put the issue number in the commit message.)

---
Issue: HPPC-90 (migrated from JIRA), created by Benjamin Johnson, resolved Nov 25 2013

**Comment by dweiss** [comment]
Comment by Benjamin Johnson (migrated from JIRA)

Fixed in https://github.com/picomancer/hppc/commit/e8630f0b80aca1a9289039451a0f2fb185702cdd

**Comment by dweiss** [comment]
Comment by Dawid Weiss (@dweiss) (migrated from JIRA)

Why do you need an Ant build script, Benjamin? Just curious. Can you create a pull request on gh? It'll save the commit as yours in git history.

**Comment by dweiss** [comment]
Comment by Dawid Weiss (@dweiss) (migrated from JIRA)

I looked at this again. Your rationale:
> The file doesn't contain anything for the template processor, so it doesn't belong in templates directory\.

is not entirely correct. The 'templates' are actually valid Java code – they should be compilable using javac without any preprocessing (that was an important goal of how I chose to implement template substitutions). And this file does logically belong to the "template" classes because it contains all sorts of definitions for capacity bumping.
